### DISCUSSION TITLE
Arrays and identifiers

### DISCUSSION THREAD

**Discussion by andrejunges** [react-hook-form/react-hook-form#1799] [Q&A] [answered]
In case I need the ids used as keys on array fields, the recommended approach is to add an input type hidden with it?

Example, I'm loading an existing category record with a child list of products:
The user can add new ones, edit and delete existing ones.
```
category {
  name
  products: [
    { id: 1, name: "..." }
  ]
}
```

In the end when I'm massaging this data to save it I'd like to know the `ID ` of the ones that are existing records, any suggestions?

**Comment by bluebill1049** [comment, marked as answer]
you mean change the default id to something else with `keyName`, and store the id in input hidden that's correct.

**Reply by andrejunges** [reply]
the `keyName` itself doesn't really matter, I just need the ID value when I submit the form.

let's say I have `category AB` with three products:
{ id: "unique-id-1", product: "A 1" }
{ id: "unique-id-2", product: "A 2" }
{ id: "unique-id-3", product: "A 3" }

Then the user load it in a form and do two things:
1. add one more product
{ id: "random-guid", product: "A 4" }

2. change the product "A 2"s name,
{ id: "unique-id-2", product: "A 2 new name" }

When the user submits this form I need the existing record ids to send to the backend.
Is adding a hidden input the only way to have these values on the submit data?

And thanks for the awesome library and quick responses.

**Reply by bluebill1049** [reply]
yes @andrejunges I think input type hidden is the cleanest and follows the HTML standard. otherwise, you will have to do custom register with setValue:
```
useEffect(() => {
  register('xxx.id')
  setValue('xxx.id', '1')
})
```
